apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -455,7 +462,14 @@ abstract class DefaultApolloExtension(
       task.warnOnDeprecatedUsages.set(service.warnOnDeprecatedUsages)
       task.failOnWarnings.set(service.failOnWarnings)
       @Suppress("DEPRECATION")
-      task.customScalarsMapping.set(service.customScalarsMapping.orElse(service.customTypeMapping))
+      task.scalarsMapping.set(
+          service.scalarMapping.orElse(
+              service.customScalarsMapping.orElse(
+                  service.customTypeMapping
+              )
+                  .getOrNull()?.asScalarInfoMapping() ?: emptyMap()
+          )
```

Comment:
throw if both `customScalarsMapping` and `sclarMapping` are set?

Reply:
Good point!